### ISSUE TITLE
Accessibility: Adjust colours to meet at least WCAG AA

### ISSUE THREAD

**Issue by artfulrobot** [civicrm/org.civicrm.shoreditch#56]
> **Note**: This was moved from an [issue on the styleguide](https://github.com/civicrm/org.civicrm.styleguide/issues/6) by request from @AkA84. Below is the original text and then I've copied the original responses below that.

----------------

I found it hard to read text in the Shoreditch theme. My eyes are not bad and I have a properly colour managed and reasonably accurate monitor that covers the full sRGB gammut. Most users have v poor screens (less contrast) and of course many people are visually impared. CiviCRM needs to be a CRM for everyone, not just those with fantastic eyesight and fancy monitors :-)

According to http://contrastchecker.com/

None of the main colour combos are accessible at the normal font size.

![screenshot from 2017-09-11 11-05-05](https://user-images.githubusercontent.com/870343/30269588-3a444b66-96e1-11e7-9020-16181487f465.png)

I'd like CiviCRM to adhere to WCAG AAA standard, but AA standard should be a minimum, in my opinion.

-----------

@JoeMurray  said:

> @colemanw @guanhuan I atrongly support these concerns. Where should issue reside?

**Comment by artfulrobot** [comment]
The contributing guidelines ask me to apply the "Bug" label to this, but I do not appear to have permissions to do that. (This is a bug not an enhancement IMO, since improving the visual UX should not be at the cost of making things worse for the visually impared.)

**Comment by AkA84** [comment]
@artfulrobot thank you for opening this issue. I wanted to clarify if you saw this problem across the board or only with specific elements (for example i think the element in your screenshot is the [alert](https://getbootstrap.com/docs/3.3/components/#alerts)), and if so, which ones?

**Comment by artfulrobot** [comment]
I started checking the alerts because I found them hard, but then I went on to check the base combo (last sample in screenshot) used on `/civicrm/styleguide/bootstrap` which is also not accessible. I'm unsure what combination of subtle greys are expected to go together.

The styleguide says **727E8A** is body copy.  Let's check that against the backgrounds:

![textbody](https://user-images.githubusercontent.com/870343/30424273-5989ffbc-993d-11e7-8b4c-5a54bbae192c.png)

And it says Headings use #4D5663. Check that...

![heading](https://user-images.githubusercontent.com/870343/30424376-aa50f1da-993d-11e7-8bd4-f870b4bc0589.png)


So the headings are much less of an issue, and are AAA compliant if they are at least 18pt (24 css px at least in Firefox) in size.

I've not checked the dark backgrounds on which white would presumably be used?

**Comment by JonScreat** [comment]
We have plans to update the colour palette used in Shoreditch to ensure that everything passes WCAG AA. 

The changes would be as follows:
Dark Grey
#4D5663   >   #464354

Mid Grey
#727E8A   >   #4D4D69

Blue
#42AFCB   >   #2574A9

Red
#E6807F   >   #CF3458

Green
#8EC68A   >   #4D994D 	(For text and icons)
#8EC68A   >   #44CB7E 	(For buttons and accents (button text must be #464354))

Orange
#ECA67F   >   #BF5900 		(For text and Icons)
#ECA67F   >   #E6AB5E 	(For buttons and accents (button text must be #464354))

A preview of this can be seen here:
https://projects.invisionapp.com/share/7QCECB68Z#/screens/254761833_WCAG_AA_Options

Feedback is welcome.

**Comment by artfulrobot** [comment]
Woop woop, this looks fantastic, *and* I can read it!

**Comment by mukeshcompucorp** [comment]
@AkA84 This is fixes as part of https://github.com/civicrm/org.civicrm.shoreditch/pull/94 we can close this one

**Comment by JoeMurray** [comment]
I've just reviewed the contrast for text in current version of shoreditch on dmaster and found it non-compliant with WCAG 2.0 AA. See https://lab.civicrm.org/dev/accessibility/issues/11#note_11457

**Comment by JoeMurray** [comment]
Should I create a separate issue as this is not related to colour choices per se but to the contrast between foreground and background colours.

**Comment by JoeMurray** [comment]
Given lack of response and the fact the issues are different though related, I have created a separate issue for contrast ratios at https://github.com/civicrm/org.civicrm.shoreditch/issues/348

**Comment by jamienovick** [comment]
@JoeMurray Can I close this issue for the time being and we can discuss the contrast separately?
